tests/testthat/test-unnest.R

```diff
@@ -167,3 +167,11 @@ test_that("grouping is preserved", {
   expect_equal(class(df), class(rs))
   expect_equal(dplyr::groups(df), dplyr::groups(rs))
 })
+
+test_that("column order is preserved", {
+  df <- tibble(x = "a", y = list("b", "c"), z = "d")
+  expect_named(unnest(df), c("x", "y", "z"))
+
+  df$y <- list(tibble(y = "b"), tibble(y = "c"))
```

Comment:
The test would be clearer if the internal tibbles had two columns with different names than `y` .

Reply:
I see what you mean. What do you think about:

```r
df$y <- list(tibble(y = "b", y2 = "e"), tibble(y = "c", y2 = "f"))
```

This verifies the order of the original columns (`x`, `y`, and `z`) is preserved and new columns from the internal tibbles (`y2`) are appended.

```r
unnest(df)
## # A tibble: 2 x 4
##   x     y     z     y2   
##   <chr> <chr> <chr> <chr>
## 1 a     b     d     e    
## 2 a     c     d     f 
```

Reply:
I'd have expected the new column order to be x, y, y2, z, same if y isn't part of the nested tibble. What's the rationale behind leaving only y in the original place?